### PR TITLE
[New resource] - `azurerm_machine_learning_datastore_datalake_gen2`

### PR DESCRIPTION
**Pull request by xuzhang3**
Support Machine learning data store - data lake gen2. This PR depends on #19909 . 
```log
=== RUN   TestAccMachineLearningDataStoreDataLakeGen2_basic
=== PAUSE TestAccMachineLearningDataStoreDataLakeGen2_basic
=== RUN   TestAccMachineLearningDataStoreDataLakeGen2_spn
=== PAUSE TestAccMachineLearningDataStoreDataLakeGen2_spn
=== RUN   TestAccMachineLearningDataStoreDataLakeGen2_Update
=== PAUSE TestAccMachineLearningDataStoreDataLakeGen2_Update
=== RUN   TestAccMachineLearningDataStoreDataLakeGen2_requiresImport
=== PAUSE TestAccMachineLearningDataStoreDataLakeGen2_requiresImport
=== CONT  TestAccMachineLearningDataStoreDataLakeGen2_basic
=== CONT  TestAccMachineLearningDataStoreDataLakeGen2_Update
=== CONT  TestAccMachineLearningDataStoreDataLakeGen2_requiresImport
=== CONT  TestAccMachineLearningDataStoreDataLakeGen2_spn
--- PASS: TestAccMachineLearningDataStoreDataLakeGen2_spn (589.94s)
--- PASS: TestAccMachineLearningDataStoreDataLakeGen2_basic (590.51s)
--- PASS: TestAccMachineLearningDataStoreDataLakeGen2_requiresImport (607.17s)
--- PASS: TestAccMachineLearningDataStoreDataLakeGen2_Update (711.76s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/machinelearning       738.965s

```

### CODE REVIEW

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @xuzhang3 
As with the other resources in this RP and mentioned offline, we need them to be using the Typed SDK.  If you can update, we can take another look.

Thanks!

**Comment by xuzhang3** [comment]
> Hi @xuzhang3 As with the other resources in this RP and mentioned offline, we need them to be using the Typed SDK. If you can update, we can take another look.
> 
> Thanks!

@jackofallops   working on it, #19909  has been changed to typed resource.

File: website/docs/r/machine_learning_datastore_datalake_gen1.html.markdown [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+* `authority_url` - (Optional) An URL used for authentication.
+
+* `description` - (Optional) Text used to describe the asset. Changing this forces a new Machine Learning DataStore to be created.
+
+* `is_default` - (Optional) Specifies whether this Machines Learning DataStore is the default for the Workspace. Defaults to `false`.
+
+~> **Note:** `is_default` can only be set to `true` on update.
+
+* `service_data_auth_identity` - (Optional) Specifies which identity to use when retrieving data from the specified source. Defaults to `None`. Possible values are `None`, `WorkspaceSystemAssignedIdentity` and `WorkspaceUserAssignedIdentity`.
```
Comment (katbyte) [inline website/docs/r/machine_learning_datastore_datalake_gen1.html.markdown:95]: isn't auth implied by identity? 

```suggestion
* `service_data_identity` - (Optional) Specifies which identity to use when retrieving data from the specified source. Defaults to `None`. Possible values are `None`, `WorkspaceSystemAssignedIdentity` and `WorkspaceUserAssignedIdentity`.
```

File: website/docs/r/machine_learning_datastore_datalake_gen1.html.markdown [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+* `client_secret` - (Optional) The secret of the Service Principal.
+
+* `authority_url` - (Optional) An URL used for authentication.
+
+* `description` - (Optional) Text used to describe the asset. Changing this forces a new Machine Learning DataStore to be created.
+
+* `is_default` - (Optional) Specifies whether this Machines Learning DataStore is the default for the Workspace. Defaults to `false`.
+
+~> **Note:** `is_default` can only be set to `true` on update.
```
Comment (katbyte) [inline website/docs/r/machine_learning_datastore_datalake_gen1.html.markdown:93]: i assume this means we cannot set it to true on creation - can we not simply do an update call at the end of the create function to set this?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/machine_learning_datastore_datalake_gen1.html.markdown (hunk 1 above)
Comment (xuzhang3) [reply website/docs/r/machine_learning_datastore_datalake_gen1.html.markdown]: Yes, indicates whether the workspace managed identity can view the data.

File: website/docs/r/machine_learning_datastore_datalake_gen1.html.markdown (hunk 2 above)
Comment (xuzhang3) [reply website/docs/r/machine_learning_datastore_datalake_gen1.html.markdown]: This property cannot be update.  It managed by another API, checking this with service team.

**Review by katbyte — APPROVED** [review]
LGTM 🧄

**Pull request merged**